src/dotnet/runner/Data/BAData.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DotnetRunner.Data
+{
+
+    // rows is nrows+1 vector containing
+    // indices to cols and vals. 
+    // rows[i] ... rows[i+1]-1 are elements of i-th row
+    // i.e. cols[row[i]] is the column of the first
+    // element in the row. Similarly for values.
+    public class BASparseMatrix
```

Comment:
Ideally we would do this in F#, or use a generic SparseMatrix class in an F# numerics library.
Do you know of one?